### PR TITLE
Adds Empty Object Type Validation Rule

### PR DESCRIPTION
**Pull request by michaelstaib**

### CODE REVIEW

File: spec/Section 4 -- Composition.md [hunk 1]
```diff
@@ -18,4 +18,66 @@ run in sequence to produce the composite execution schema.
 
 ### Post Merge Validation
 
+#### Empty Merged Object Type
+
+**Error Code**
+
+`Empty_Merged_Object_Type`
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:25]:
```suggestion
`EMPTY_MERGED_OBJECT_TYPE`
```

File: spec/Section 4 -- Composition.md [hunk 2]
```diff
@@ -18,4 +18,66 @@ run in sequence to produce the composite execution schema.
...
+**Formal Specification**
+
+- Let {types} be the set of all object types across all source schemas
+- For each {type} in {types}:
+  - {IsObjectTypeEmpty(type)} must be false.
+
+IsObjectTypeEmpty(type):
+
+- Let {fields} be a set of all fields of all types with coordinate and kind
+  {type} across all source schemas
```
Comment (glen-84) [inline spec/Section 4 -- Composition.md:36]: I'm not following the "with coordinate and kind {type}" part.

**Review by glen-84 — CHANGES_REQUESTED** [review]

File: spec/Section 4 -- Composition.md (hunk 1 above)
Comment (PascalSenn) [reply spec/Section 4 -- Composition.md:25]: done

File: spec/Section 4 -- Composition.md (hunk 2 above)
Comment (PascalSenn) [reply spec/Section 4 -- Composition.md:36]: done

**Pull request merged**